### DISCUSSION TITLE
Export an api for differential forms?

### DISCUSSION THREAD

**Discussion by Antoinemarteau** [mikeingold/LineIntegrals.jl#120] [General]
I understand the package leverages differential calculus internally to compute integral of correct unit over domain of correct dimension, do you have any plan to export an API for a differential form defined on a mesh/domain to enable doing computation with them before integrating ?

**Comment by JoshuaLampert** [comment]
Hi @Antoinemarteau! Thanks for your interest. Yes, we do have plans to expand the support for dealing with differential forms. Recently, @kylebeggs continued the initiative and started looking into switching from finite difference formulas to compute the differential to automatic differentiation. See #35 for an issue tracking the progress. We plan to upstream an API for differential forms to Meshes.jl (see also https://github.com/JuliaGeometry/MeshIntegrals.jl/issues/115#issuecomment-2432595573 for our plan). Until we finished this effort, you can already use the exported function `jacobian(geometry, ts; ε=1e-6)` (see the [documentation](https://juliageometry.github.io/MeshIntegrals.jl/stable/api/#Derivatives) for more info) and the non-exported function `differential(geometry, ts)`, see https://github.com/JuliaGeometry/MeshIntegrals.jl/blob/6c8cd1f74cec0715d7255d60e48f57f202af311d/src/differentiation.jl#L85-L90
As mentioned above, these use a central finite difference formula (and an analytical derivative for the special case of a `BezierCurve`).
